src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GVMDependenciesNode.cs

```diff
@@ -41,14 +41,42 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
             // TODO: https://github.com/dotnet/corert/issues/3224
             if (_method.IsAbstract)
             {
-                return new DependencyListEntry[]
-                {
-                    new DependencyListEntry(context.ReflectableMethod(_method), "Abstract reflectable method"),
-                };
+                yield return new DependencyListEntry(context.ReflectableMethod(_method), "Abstract reflectable method");
             }
+            else
+            {
+                // Universal canonical instantiations should be entirely universal canon
+                if (_method.IsCanonicalMethod(CanonicalFormKind.Universal))
+                    _method = _method.GetCanonMethodTarget(CanonicalFormKind.Universal);
```

Comment:
Could you mark `_method` readonly so that that it doesn't invite mutating state in such unpredictable ways?

Why do we even allow creating a `GVMDependenciesNode` like this? Shouldn't these be fully canonicalized?